packages/ui/FileInput/index.tsx

```diff
@@ -1,213 +1,377 @@
+      if (file.status === "fulfilled") {
+        uploadedUrls.push(file.value.url);
+      }
+    });
+
+    const prevUrls = Array.isArray(fileUrl) ? fileUrl : fileUrl ? [fileUrl] : [];
+    onFileUpload([...prevUrls, ...uploadedUrls]);
   };
 
+  useEffect(() => {
+    const fs = () => {
+      if (fileUrl && typeof fileUrl === "string") {
+        return [{ url: fileUrl, name: fileUrl.split("/").pop() || "", uploaded: true }];
+      } else if (fileUrl && Array.isArray(fileUrl)) {
+        return fileUrl.map((url) => ({ url, name: url.split("/").pop() || "", uploaded: true }));
+      } else {
+        return [];
+      }
+    };
+    setFiles(fs());
+  }, [fileUrl]);
+
   return (
-    <label
-      htmlFor="selectedFile"
-      className={cn(
-        "relative flex h-52 w-full cursor-pointer flex-col items-center justify-center rounded-lg border-2 border-dashed border-slate-300 bg-slate-50 hover:bg-slate-100 dark:border-slate-600 dark:bg-slate-700 dark:hover:border-slate-500 dark:hover:bg-slate-600 dark:hover:bg-slate-800",
-        isError && "border-red-500"
-      )}
-      onDragOver={(e) => handleDragOver(e)}
-      onDrop={(e) => handleDrop(e)}>
-      {isUploaded && fileUrl ? (
-        <>
-          <div className="absolute inset-0 mr-4 mt-2 flex items-start justify-end gap-4">
-            <div className="flex h-8 w-8 items-center justify-center rounded-lg bg-slate-300 bg-opacity-50 text-slate-800 hover:bg-slate-200/50 hover:text-slate-900">
-              <label htmlFor="modifyFile">
-                <PhotoIcon className="h-5 cursor-pointer text-slate-700 hover:text-slate-900" />
-
-                <input
-                  type="file"
-                  id="modifyFile"
-                  name="modifyFile"
-                  accept={allowedFileExtensions.map((ext) => `.${ext}`).join(",")}
-                  className="hidden"
-                  onChange={async (e) => {
-                    const file = e.target?.files?.[0];
-                    if (file) {
-                      await handleFileUpload({
-                        file,
-                        allowedFileExtensions,
-                        environmentId,
-                      });
-                    }
-                  }}
-                />
-              </label>
-            </div>
-            <div className="flex h-8 w-8 items-center justify-center rounded-lg bg-slate-300 bg-opacity-50 hover:bg-slate-200/50">
-              <TrashIcon
-                className="h-5 text-slate-700 hover:text-slate-900"
-                onClick={() => onFileUpload(undefined)}
-              />
-            </div>
-          </div>
+    <div className="w-full cursor-default">
+      {files.length > 0 ? (
+        multiple ? (
+          <div className="flex flex-wrap gap-2">
+            {files.map((file, idx) => (
+              <>
+                {file.name.endsWith("jpg") || file.name.endsWith("jpeg") || file.name.endsWith("png") ? (
```

Comment:
please use a different array for handing the allowed file types for preview, you could make an array like
`allowedFileTypesForPreview` and use a `includes` on it to do this, also, please include the extension `webp` in the `allowedFileExtensions` type as well as here 🙂